src/server/worker/pipeline/transform/worker.go

```diff
@@ -377,6 +377,8 @@ func handleDatumSet(driver driver.Driver, logger logs.TaggedLogger, datumSet *Da
 								return errors.EnsureStack(driver.RunUserErrorHandlingCode(runCtx, logger, env))
 							}))
 						}
+						// maybe proto/clone meta here
```

Comment:
We should probably go ahead and do the clone. We could add `meta = proto.Clone(meta).(*datum.Meta)` as the first line in the callback.